OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs

```diff
@@ -300,14 +303,32 @@ void StartMissionClicked()
 				PlayVideo(fsPlayer, gameStartVideo, PlayingVideo.GameStart, () =>
 				{
 					StopVideo(fsPlayer);
-					Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
+					StartMission();
 				});
 			}
 			else
-				Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
-		}
+				StartMission();
+        }
+
+		void StartMission()
+		{
+			OrderManager om = null;
+
+			Action lobbyReady = null;
+			lobbyReady = () =>
```

Comment:
I'm suggesting something like the following:
```csharp
public static void CreateAndStartLocalServer(string mapUID, IEnumerable<Order> setupOrders, Action onStart = null)
{
	OrderManager om = null;

	Action lobbyReady = null;
	lobbyReady = () =>
	{
		LobbyInfoChanged -= lobbyReady;
		foreach (var o in setupOrders)
			om.IssueOrder(o);

		if (onStart != null)
			onStart();

		om.IssueOrder(Order.Command("startgame"));
	};
	LobbyInfoChanged += lobbyReady;

	om = JoinServer(IPAddress.Loopback.ToString(), CreateLocalServer(mapUID), "");
}
```

Which is then called like:
```csharp
var orders = new[] { Order.Command("gamespeed {0}".F(gameSpeed)), Order.Command("difficulty {0}".F(difficulty)) };
Game.StartLocalGame(selectedMapPreview.Uid, orders, onStart);
```
or 
```csharp
Game.StartLocalGame(selectedMapPreview.Uid, new[] { Order.Command("sync_lobby {0}".F(lobbyInfo.Serialize()) }, onStart);
```